File: Data Structures & Algorithms/insert-new-interval/submission-4.py
class Solution:
    def insert(self, intervals: List[List[int]], newInterval: List[int]) -> List[List[int]]:
        intervals.append(newInterval)
        intervals.sort(key=lambda x:x[0])
        res=[]

        prevInterval=intervals[0]
        for currInterval in intervals[1:]:
            start,end=currInterval
            if start<=prevInterval[1]:
                prevInterval=[prevInterval[0], max(prevInterval[1],end)]
            else:
                res.append(prevInterval)
                prevInterval=currInterval
        res.append(prevInterval)
        return res

File: Data Structures & Algorithms/non-overlapping-intervals/submission-4.java
class Solution {
    public int eraseOverlapIntervals(int[][] intervals) {
        Arrays.sort(intervals, (a,b)->Integer.compare(a[0],b[0]));
        int minRemovals=0;
        int[] prevInterval=intervals[0];
        for(int i=1; i<intervals.length; i++){
            int start=intervals[i][0];
            int end=intervals[i][1];
            if(start<prevInterval[1]){
                minRemovals+=1;
                prevInterval[1]=Math.min(prevInterval[1], end);
            }
            else{
                prevInterval=intervals[i];
            }
        }
        return minRemovals;
    }
}
